plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedView.java

```diff
@@ -86,6 +86,189 @@ protected String getStorageMetadataLocation(String materializedViewName)
         return table.getParameters().get(METADATA_LOCATION_PROP);
     }
 
```

Comment:
please also add a test with `update/delete`. Part of scenario would be that incremental refresh should work again after compaction.

Reply:
Added a new test case which verifies that we fall back to FULL refresh for cases when delete files are present. As for compaction reenabling incremental refresh, please see my comment below

Reply:
> As for compaction reenabling incremental refresh, please see my comment below

Would a test like that work:
1. initial (full) refresh with some data
2. add one row to source table
3. incremental refresh
6. compaction
7. full refresh
8. add one row to source table
9. we are back to incremental refresh

Reply:
This is good, I would just modify step 4:
1. initial (full) refresh with some data
2. add one row to source table
3. incremental refresh
4. update or delete one row
5. full refresh
6. add one row to source table
7. we are back to incremental refresh

Reply:
In the original test case you proposed, step 5 shouldn't do any refresh (essentially a noop), because step 4 (compaction) didn't add any new rows, it just binpacked the existing rows into fewer files.